substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/config/SerializationConfiguration.java

```diff
@@ -0,0 +1,85 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.configure.config;
+
+import com.oracle.svm.configure.json.JsonPrintable;
+import com.oracle.svm.configure.json.JsonWriter;
+import com.oracle.svm.core.configure.SerializationKey;
+
+import java.io.IOException;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class SerializationConfiguration implements JsonPrintable {
+
+    private final ConcurrentHashMap.KeySetView<SerializationKey<String>, Boolean> serializations = ConcurrentHashMap.newKeySet();
+
+    public void add(String serializationTargetClass, String[] parameterTypes, String[] checkedExceptions, int modifiers, String targetConstructorClass) {
+        SerializationKey<String> key = new SerializationKey<>(serializationTargetClass, parameterTypes, checkedExceptions, modifiers, targetConstructorClass);
+        add(key);
+    }
+
+    public void add(SerializationKey<String> key) {
+        serializations.add(key);
+    }
+
+    public boolean contains(String serializationTargetClass, String[] parameterTypes, String[] checkedExceptions, int modifiers, String targetConstructorClass) {
+        SerializationKey<String> key = new SerializationKey<>(serializationTargetClass, parameterTypes, checkedExceptions, modifiers, targetConstructorClass);
+        return serializations.contains(key);
+    }
+
+    @Override
+    public void printJson(JsonWriter writer) throws IOException {
+        writer.append('[').indent();
+        String prefix = "";
+        for (SerializationKey<String> skey : serializations) {
+            writer.append(prefix).newline().append('{');
+            writer.newline();
+            String typePrefix = "";
+            writer.quote("name").append(":").quote(skey.getSerializationTargetClass()).append(",").newline();
+            writer.quote("parameterTypes").append(":").append('[');
+            for (String parameterType : skey.getParameterTypes()) {
+                writer.append(typePrefix).quote(parameterType);
+                typePrefix = ",";
+            }
+            writer.append(']').append(",").newline();
+
+            typePrefix = "";
+            writer.quote("checkedExceptions").append(":").append('[');
+            for (String checkedException : skey.getCheckedExceptions()) {
+                writer.append(typePrefix).quote(checkedException);
+                typePrefix = ",";
+            }
+            writer.append(']').append(",").newline();
+
+            writer.quote("modifiers").append(':').append(Integer.toString(skey.getModifiers())).append(",").newline();
+            writer.quote("targetConstructorClass").append(':').quote(skey.getTargetConstructorClass()).newline();
+            writer.append('}');
```

Comment:
Couldn't you make `SerializationKey` create its `flattenString` in JSON format and have it implement `JsonPrintable`?

Reply:
Fixed. But `SerializationKey` is located in `com.svm.core` which should not have dependency on `com.oracle.svm.configure` where  `JsonPrintable` is located, so I created a sub class of  `SerializationKey` in `com.oracle.svm.configure`  to print json.